microsoft/testsuites/dpdk/dpdkperf.py

```diff
@@ -0,0 +1,242 @@
+from lisa.messages import NetworkPPSPerformanceMessage, create_message
+from lisa.tools import Lscpu
+from lisa.util import constants
+from microsoft.testsuites.dpdk.dpdkutil import (
+    DpdkTestResources,
+    verify_dpdk_send_receive,
+    verify_dpdk_send_receive_multi_txrx_queue,
+)
+
+
+@TestSuiteMetadata(
+    area="dpdk",
+    category="performance",
+    description="""
+    This test suite is to validate DPDK performance
+    """,
+)
+class DpdkPerformance(TestSuite):
+    TIMEOUT = 12000
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: failsafe pmd
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_failsafe_dual_core(
+        self,
+        environment: Environment,
+        log: Logger,
+        variables: Dict[str, Any],
+    ) -> None:
+        self._run_dpdk_perf_test("failsafe", environment, log, variables)
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: failsafe pmd
+        Uses all available cores and default queues
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_failsafe_multi_core(
+        self,
+        environment: Environment,
+        log: Logger,
+        variables: Dict[str, Any],
+    ) -> None:
+
+        core_counts = [
+            n.tools[Lscpu].get_core_count() for n in environment.nodes.list()
+        ]
+
+        assert_that(core_counts).described_as(
+            "Nodes contain different core counts, DPDK Suite expects sender "
+            "and receiver to have same core count."
+        ).contains_only(core_counts[0])
+
+        self._run_dpdk_perf_test(
+            "failsafe", environment, log, variables, use_cores=core_counts[0]
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: failsafe pmd
+        uses multiple cores and tx/rx queues
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_failsafe_multi_queue(
+        self,
+        environment: Environment,
+        log: Logger,
+        variables: Dict[str, Any],
+    ) -> None:
+
+        self._run_dpdk_perf_test(
+            "failsafe", environment, log, variables, use_queues=True
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: netvsc pmd
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_netvsc_dual_core(
```

Comment:
The requirement shows "Sriov", but the test case name has "netvsc" as synthetic. Is it expected? If it's synthetic, the name "perf_dpdk_synthetic_dual_core" is better.

Reply:
The naming conventions are confusing. I used the same ones as V2 but they're confusing. Maybe we should just change them here. Failsafe bonds to the synthetic interface which is paired with the VF so it can fail over. 'Netvsc' bonds to the VF itself but can experience interruptions if the VF has an issue.

Maybe we can change to 'failsafe' and 'vf_direct' ?

Reply:
testpmd uses those terms as arguments but we can still change the test names to be more clear

Reply:
Ah that's right, those are the names of the DPDK PMD's. Maybe we should leave them actually, even though when you dig through the code it can be confusing because there are tons of references to the actual driver hv_netvsc.

Reply:
https://doc.dpdk.org/guides-20.11/nics/netvsc.html
https://doc.dpdk.org/guides-20.11/nics/fail_safe.html

Reply:
I see, thank you for the documents! let's use consistent names like dpdk.

Reply:
I'll change the test names to failsafe_pmd and netvsc_pmd instead of netvsc and failsafe